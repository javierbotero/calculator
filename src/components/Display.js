import React from 'react';
import PropTypes from 'prop-types';

export default function Display({ result }) {
  let str;
  if (result.total === null) {
    str = '0';
  } else {
    str = `${result.total}
 ${result.operation ? result.operation : ''}
 ${result.next ? result.next : ''}`;
  }
  console.log(result);
  return <div>{str}</div>;
}

Display.defaultProps = {
  result: '0',
};

Display.propTypes = {
  result: PropTypes.string,
};
