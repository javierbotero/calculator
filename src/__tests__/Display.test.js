import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Display from '../components/Display';

describe('Tests Display', () => {
  it('Makes sure display has the text "100 / 5" when the prop result has the object with that corresponding values', () => {
    const object = {
      total: '100',
      next: '5',
      operation: '/',
    };
    render(<Display result={object} />);
    const div = screen.getByText('100', { exact: false });
    expect(div).toHaveTextContent('100 / 5');
  });

  it('Prints 0 when total has null value', () => {
    const object = {
      total: null,
      next: null,
      operation: null,
    };

    render(<Display result={object} />);
    expect(screen.getByText('0')).toHaveTextContent('0');
  });

  it('Checks the component match with th snapshot', () => {
    const object = {
      total: null,
      next: null,
      operation: null,
    };
    const tree = renderer
      .create(<Display result={object} />)
      .toJSON()
    expect(tree).toMatchSnapshot();
  });
});
