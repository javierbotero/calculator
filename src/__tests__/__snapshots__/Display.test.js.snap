// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Display Checks the component match with th snapshot 1`] = `
<div
  className="display"
>
  0
</div>
`;
